front/src/applications/stdcm/hooks/useStdcm.ts

```
@@ -87,75 +117,183 @@ const useStdcm = ({
     }
   };
 
-  const launchStdcmRequest = async () => {
-    setStdcmResponse(undefined);
-    setStdcmTrainConflicts(undefined);
+  const handleSuccess = (
+    response: StdcmApiSuccessResponse,
+    payload: PostTimetableByIdStdcmApiArg
+  ) => {
+    if (!stdcmRollingStock) return;
```

This should never happen, so you can either raise an error here or use ! where the stdcmRollingStock is used

---

Done !